store/kv/kv.go

```diff
@@ -167,14 +332,132 @@ func (k *KeyValueStoreImpl) BeginTx(ctx context.Context) (Tx, error) {
 	}, nil
 }
 
-func (k *KeyValueStoreImpl) GetInternalDatabase() interface{} {
-	return k.db
+func (m *KeyValueStoreImplWithMetrics) BeginTx(ctx context.Context) (Tx, error) {
+	// This needs to be a special case in order to have the tx metrics as well
+	var btx Tx
+	var err error
+	m.measure("BeginTx", func() error {
+		btx, err = m.kv.BeginTx(ctx)
+		return err
+	})
+
+	return &TxImplWithMetrics{
+		btx,
+	}, err
+
+}
+
+func (k *KeyValueStoreImpl) GetInternalDatabase() (interface{}, error) {
+	return k.db, nil
+}
+
+func (m *KeyValueStoreImplWithMetrics) GetInternalDatabase() (interface{}, error) {
+	var k interface{}
+	var err error
+	m.measure("GetInternalDatabase", func() error {
+		k, err = m.kv.GetInternalDatabase()
+		return err
+	})
+	return k, err
 }
 
 type TxImpl struct {
 	*ftx
 }
 
+type TxImplWithMetrics struct {
+	tx Tx
+}
+
+func (m *TxImplWithMetrics) measure(name string, f func() error) {
+	tags := metrics.GetFdbReqTags(name, true)
+	defer metrics.FdbRequests.Tagged(tags).Histogram("histogram", tally.DefaultBuckets).Start().Stop()
```

Comment:
the difference with non-tx version is just tags.
so you can just create a helper method `measureLow(name, f, isTx)` and use it in both places.
```
func (m *TxImplWithMetrics) measure(name string, f func() error) {
	measureLow(name, f, true)
}
```
```
func (m *KeyValueStoreImplWithMetrics) measure(name string, f func() error) {
	measureLow(name, f, false)
}
```

Reply:
This is great, also much easier for testing. changed it.